### PR TITLE
Add Transaction Capabilities.

### PR DESCRIPTION
**Pull request by tusharmath**

### CODE REVIEW

**Comment by iravid** [comment]
@tusharmath Hey. This is really cool. I need to find 20 minutes to play around with it in the console, looks good beyond that! Hope to do that tonight.

File: src/main/scala/zio/rocksdb/TransactionDB.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package zio.rocksdb
+
+import org.{ rocksdb => jrocks }
+import zio._
+
+object TransactionDB extends Operations[TransactionDB, service.TransactionDB] {
+  private final class Live(db: jrocks.TransactionDB) extends RocksDB.Live(db, Nil) with service.TransactionDB {
```
Comment (iravid) [inline src/main/scala/zio/rocksdb/TransactionDB.scala]: The `Nil` passed to the `RocksDB.Live` constructor doesn't look right. Are we saying that `TransactionDB` only supports the default column family?

File: src/main/scala/zio/rocksdb/service/Transaction.scala [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package zio.rocksdb.service
+
+import org.{ rocksdb => jrocks }
+import zio.{ Task, UIO }
+
+trait Transaction {
+  def get(readOptions: jrocks.ReadOptions, key: Array[Byte]): Task[Option[Array[Byte]]]
```
Comment (iravid) [inline src/main/scala/zio/rocksdb/service/Transaction.scala:11]: Can we add scaladoc to this interface?

File: src/main/scala/zio/rocksdb/service/TransactionDB.scala [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package zio.rocksdb.service
+
+import org.{ rocksdb => jrocks }
+import zio.{ Has, UIO, ZIO }
+
+trait TransactionDB extends RocksDB {
+  def beginTransaction(writeOptions: jrocks.WriteOptions): UIO[Transaction]
```
Comment (iravid) [inline src/main/scala/zio/rocksdb/service/TransactionDB.scala]: Can we add scaladoc to this interface?

**Review by iravid — CHANGES_REQUESTED** [review]
Looks great. I like this structure!

In addition to the comments I left, because the interface was changed (`rocksdb => RocksDB` for the accessors), would be great if you could update the `README.md`.

File: src/main/scala/zio/rocksdb/TransactionDB.scala (hunk 1 above)
Comment (tusharmath) [reply src/main/scala/zio/rocksdb/TransactionDB.scala]: No. I think we can add that capability later. I though having transactional capabilities first and stabilizing the API right now would be more important than increase it's surface area. 

There are multiple other variants for other functions that are available but I have refrained my self from adding them either.

Comment (iravid) [reply src/main/scala/zio/rocksdb/TransactionDB.scala]: Sounds good!

File: project/metals.sbt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+// DO NOT EDIT! This file is auto-generated.
+// This file enables sbt-bloop to create bloop config files.
+
+addSbtPlugin("ch.epfl.scala" % "sbt-bloop" % "1.4.0-RC1-190-ef7d8dba")
```
Comment (tusharmath) [inline project/metals.sbt:4]: Let me know if you think I should delete this file. I use VSCode for scala work so thought of adding it :)

Comment (iravid) [reply project/metals.sbt:4]: Metals adds this as part of the installation in the project, so let's remove it. It's usually good to add `project/metals.sbt` to your personal global `.gitignore`.

File: src/main/scala/zio/rocksdb/Atomically.scala [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package zio.rocksdb
+
+object Atomically {
+  object TransactionOnly { implicit val transactionOnly = this }
```
Comment (iravid) [inline src/main/scala/zio/rocksdb/Atomically.scala:4]: Please annotate the return type of the implicits

File: src/main/scala/zio/rocksdb/Atomically.scala [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+  type TransactionOnly = TransactionOnly.type
+  object TransactionWithSomething { implicit val transactionWithSomething = this }
```
Comment (iravid) [inline src/main/scala/zio/rocksdb/Atomically.scala:6]: Please annotate the return type of the implicits

**Review by iravid — APPROVED** [review]
Looks good! A few small comments that can be addressed in a follow-up.

Thank you for this awesome contribution @tusharmath!

**Pull request merged**